crates/bevy_ecs/src/reflect/resource.rs

```diff
@@ -164,24 +176,22 @@ impl ReflectResource {
     }
 }
 
-impl<C: Resource + Reflect + FromWorld> FromType<C> for ReflectResource {
+impl<C: Resource + Reflect + FromReflect> FromType<C> for ReflectResource {
```

Comment:
As a side note, did we want to change `C` to `R` while we're at it? Or maybe we handle that another time?

Reply:
Yeah I'd prefer that change, but no strong feelings on this PR vs others.

Reply:
I was also surprised when copy-pasting from the implementation of `ReflectComponent` worked without having to change the mentions of `C` in turbofish.